apps/shop/serializers.py

```diff
@@ -0,0 +1,50 @@
+from rest_framework import serializers
+
+from .models import (
+    Category,
+    Order,
+    Product,
+)
+
+
+class ProductSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Product
+        fields = '__all__'
+
+
+class CategorySerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Category
+        fields = ('id', 'name', 'parent')
+
+
+class ProductCountSerializer(serializers.Serializer):
+
+    product_id = serializers.IntegerField(required=True)
+    product_count = serializers.IntegerField(required=True)
+    product_name = serializers.CharField(source='product.name', read_only=True)
+
+    def validate(self, attrs):
+        attrs = super().validate(attrs)
+
+        if attrs['product_count'] == 0:
+            raise serializers.ValidationError(
+                'product_count must not be == 0'
+            )
+
+        return attrs
+
+
+class OrderSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Order
+        fields = ('order_id', 'user_id', 'user_email', 'status', 'products')
+
+    order_id = serializers.IntegerField(source='id')
+    user_id = serializers.IntegerField(source='user.id')
```

Comment:
The `Source` param is redundant. Remove them and explain why in reply, please.

Reply:
Since `user` is a ForeignKey field of `Order` model, `Order` objects have `user_id` field as well, wich allows to only specify `user_id` in `Meta.fields`.